library/std/src/io/mod.rs

```diff
@@ -2110,7 +2110,8 @@ pub trait BufRead: Read {
     }
 
     /// Read all bytes until a newline (the `0xA` byte) is reached, and append
-    /// them to the provided buffer.
+    /// them to the provided buffer. You do not need to clear the buffer before
+    /// appending.
```

Comment:
I think this makes it sound more confusing than before.

> You do not need to clear the buffer before appending.

This makes it sound like you don't need to clear it, because it happens automatically.

Note that https://github.com/rust-lang/book/issues/2574 is about the Rust Book, not about the standard library documentation. The std documentation before this change already clearly states that this function appends the data.